lightningd/channel_control.c

```diff
@@ -117,6 +117,31 @@ static void peer_got_funding_locked(struct channel *channel, const u8 *msg)
 		lockin_complete(channel);
 }
 
+static void peer_got_announcement(struct channel *channel, const u8 *msg)
+{
+	secp256k1_ecdsa_signature remote_ann_node_sig;
+	secp256k1_ecdsa_signature remote_ann_bitcoin_sig;
+
+	if(!(channel->channel_flags & CHANNEL_FLAGS_ANNOUNCE_CHANNEL)) {
+		channel_internal_error(channel,
+				       "No ANNOUNCE flag, no announcemnt signatures! %s",
+				       tal_hex(tmpctx, msg));
```

Comment:
This check, while correct, may be a bit aggressive in the current network and might result in a lot of failures (nodes usually just spray you with the signatures without checking the announcement flags). We can still enforce that we don't announce by simply not sending sigs ourselves.

Please don't fail the channel over such a minor offence, but log it and we can tighten this later.

Reply:
Thank you! Now I remove this check